sqlmesh/utils/metaprogramming.py

```diff
@@ -278,7 +278,8 @@ def walk(obj: t.Any) -> None:
     if name not in env:
         # We only need to add the undecorated code of @macro() functions in env, which
         # is accessible through the `__wrapped__` attribute added by functools.wraps
-        env[name] = getattr(obj, "__wrapped__", obj)
+        env[name] = obj.__wrapped__ if getattr(obj, "is_macro", None) else obj
```

Comment:
I think this section is more robust this way, because whilst `walk`ing a callable we may encounter a user function that's decorated with `wraps` and isn't a `macro`, so with previous approach we'd "drop" the decorator by only adding the original function into the `env` (due to setting `env[name]` to its `__wrapped__` attribute if present).